generators/example/src/main/scala/RocketConfigs.scala

```diff
@@ -9,149 +9,191 @@ import freechips.rocketchip.config.{Config}
 // --------------
 
 class RocketConfig extends Config(
-  new WithTop ++                                           // use default top
+  new WithTSI ++                                           // use TSI bringup
+  new WithNoGPIO ++                                        // no GPIO pins
   new WithBootROM ++                                       // use default bootrom
   new WithUART ++                                          // add a UART
+  new freechips.rocketchip.subsystem.WithNoMMIOPort ++     // no MMIO master port
+  new freechips.rocketchip.subsystem.WithNoSlavePort ++    // no MMIO slave port
   new freechips.rocketchip.subsystem.WithInclusiveCache ++ // use Sifive L2 cache
   new freechips.rocketchip.subsystem.WithNBigCores(1) ++   // single rocket-core
   new freechips.rocketchip.system.BaseConfig)              // "base" rocketchip system
 
 class HwachaRocketConfig extends Config(
-  new WithTop ++
+  new WithTSI ++
+  new WithNoGPIO ++
   new WithBootROM ++
   new WithUART ++
+  new freechips.rocketchip.subsystem.WithNoMMIOPort ++
+  new freechips.rocketchip.subsystem.WithNoSlavePort ++
   new freechips.rocketchip.subsystem.WithInclusiveCache ++
   new hwacha.DefaultHwachaConfig ++                        // use Hwacha vector accelerator
   new freechips.rocketchip.subsystem.WithNBigCores(1) ++
   new freechips.rocketchip.system.BaseConfig)
 
 // DOC include start: GemminiRocketConfig
 class GemminiRocketConfig extends Config(
-  new WithTop ++
+  new WithTSI ++
+  new WithNoGPIO ++
   new WithBootROM ++
   new WithUART ++
+  new freechips.rocketchip.subsystem.WithNoMMIOPort ++
+  new freechips.rocketchip.subsystem.WithNoSlavePort ++
   new freechips.rocketchip.subsystem.WithInclusiveCache ++
   new gemmini.DefaultGemminiConfig ++                        // use Gemmini systolic array GEMM accelerator
   new freechips.rocketchip.subsystem.WithNBigCores(1) ++
   new freechips.rocketchip.system.BaseConfig)
 // DOC include end: GemminiRocketConfig
 
 class RoccRocketConfig extends Config(
-  new WithTop ++
+  new WithTSI ++
+  new WithNoGPIO ++
   new WithBootROM ++
   new WithUART ++
+  new freechips.rocketchip.subsystem.WithNoMMIOPort ++
+  new freechips.rocketchip.subsystem.WithNoSlavePort ++
   new freechips.rocketchip.subsystem.WithInclusiveCache ++
   new freechips.rocketchip.subsystem.WithRoccExample ++    // use example RoCC-based accelerator
   new freechips.rocketchip.subsystem.WithNBigCores(1) ++
   new freechips.rocketchip.system.BaseConfig)
 
 // DOC include start: JtagRocket
 class jtagRocketConfig extends Config(
-  new WithDTMTop ++                                        // use top with dtm
-  new freechips.rocketchip.subsystem.WithJtagDTM ++        // add jtag+DTM module to coreplex
+  new WithDTM ++                                           // use top with dtm
+  new WithNoGPIO ++
   new WithBootROM ++
   new WithUART ++
+  new freechips.rocketchip.subsystem.WithJtagDTM ++        // add jtag+DTM module to coreplex
```

Comment:
Should we consider rolling this into the other mixin?
We could have `WithDTM` and `WithJTAGDTM`?

Reply:
Then you'll have duplication of code for connecting the DTM signals in the testharness. I think these two mixins do sufficiently different things that its best to leave them separate.
`WithDTM` instantiates the debug module with the DMI protocol, while `WithJtagDTM` adds JTAG to the Debug module's set of supported protocols.

Reply:
I suppose the inline comment can be corrected

Reply:
Ok thats fine. Just wanted to consider it for a bit.